managed/src/main/java/com/yugabyte/yw/controllers/AvailabilityZoneController.java

```diff
@@ -86,7 +68,8 @@ public Result create(UUID customerUUID, UUID providerUUID, UUID regionUUID) {
     } catch (Exception e) {
       LOG.error(e.getMessage());
       AvailabilityZoneData failedAz = azDataList.get(availabilityZones.size());
-      return ApiResponse.error(INTERNAL_SERVER_ERROR, "Unable to create zone: " + failedAz.code);
+      throw new YWServiceException(
```

Comment:
AvailabilityZone.createOrThrow should create AZ, catch any exceptions, log them and throw internal error. Keeps controller simple.

Reply:
Also add e.getMessage to the error  message.